sdk/cloudmachine/Azure.CloudMachine/src/CloudMachineWorkspace.cs

```diff
@@ -42,20 +39,20 @@ public CloudMachineWorkspace(TokenCredential credential = default, IConfiguratio
         {
             Credential = credential;
         }
-
-        string cmid;
-        if (configuration == default)
-        {
-            cmid = ReadOrCreateCmid();
-        }
         else
         {
-            cmid = configuration["CloudMachine:ID"];
-            if (cmid == null)
-                throw new Exception("CloudMachine:ID configuration value missing");
+            Credential = Environment.GetEnvironmentVariable("CLOUDMACHINE_MANAGED_IDENTITY_CLIENT_ID") switch
```

Comment:
does this mean we now require this env var? why is this better than the config?

Reply:
Yes, but the environment is defined as part of the AppService bicep so we control both sides.